catalog/dags/es/create_new_es_index/create_new_es_index_types.py

```diff
@@ -20,29 +20,29 @@ class CreateNewIndex:
 
     Required Constructor Arguments:
 
-    environment:     str representation of the environment in which to create
-                     the new index
-    blocking_dags:   list of dags with which to prevent concurrency; the
-                     generated create_new_es_index dag will fail immediately if
-                     any of these dags are running.
-    reindex_timeout: timedelta expressing maximum amount of time the reindexing
-                     step may take
+    environment:         str representation of the environment in which to
+                         create the new index
+    blocking_dags:       list of dags with which to prevent concurrency; the
+                         generated create_new_es_index dag will fail
+                         immediately if any of these dags are running.
+    reindex_timeout:     timedelta expressing maximum amount of time the
+                         reindexing step may take
+    requests_per_second: number of requests to send per second during ES
+                         reindexing, used to throttle the reindex step
     """
 
     dag_id: str = field(init=False)
     es_host: str = field(init=False)
     environment: str
     blocking_dags: list
+    requests_per_second: int | None = None
     reindex_timeout: timedelta = timedelta(hours=12)
 
     def __post_init__(self):
         self.dag_id = f"create_new_{self.environment}_es_index"
 
-        # Get the appropriate connection information for this environment.
-        conn = Connection.get_connection_from_secrets(
-            f"elasticsearch_http_{self.environment}"
-        )
-        self.es_host = conn.host
+        if not self.requests_per_second:
+            self.requests_per_second = Variable.get("ES_INDEX_THROTTLING_RATE")
```

Comment:
I _think_ in order for this to be an `int`, we need to add `deserialize_json=True` to the `.get` call:

```python
In [2]: Variable.get("ES_INDEX_THROTTLING_RATE")
Out[2]: '20000'

In [3]: Variable.get("ES_INDEX_THROTTLING_RATE", deserialize_json=True)
Out[3]: 20000
```

Reply:
Added to be safe, although I tested this (including checking the `type` in the tasks) and it worked. I also added a default so there will not be errors if folks don't update their .env.